src/util/db/sqlselectiterator.h

```diff
@@ -0,0 +1,46 @@
+#ifndef MIXXX_SQLSELECTITERATOR_H
+#define MIXXX_SQLSELECTITERATOR_H
+
+
+#include "util/db/fwdsqlquery.h"
+#include "util/db/sqlqueryfinisher.h"
+
+#include "util/assert.h"
+
+
+// Iterate (unidirectional and only once) over the results of a query.
+class SqlSelectIterator {
```

Comment:
I would prefer to split this class into two Iterator + range. 
The range hold the query itself and the iterator only hod a reference or likewise const reference to it.

I think "range" is the c++ name for it. In c# it is called IEnumerable.

This allow us to use for(:) as usual.  

It might be possible to reuse the QT macro 
https://github.com/qt/qtbase/blob/6bceb4a8a9292ce9f062a38d6fe143460b54370e/src/corelib/tools/qiterator.h